#560 raze - Fixed addon selection for Megaton Edition

VortexAcherontic opened the pull request:
### Common Code Submissions

* [x] Have you verified that the changes are isolated to the features or fixes you are wanting to do?
* [x] Have you tested at least one of the engines to ensure that your changes do not break existing workflow?
* [x] Have you described what your changes are accomplishing? 

While trying raze and eduke32 on Duke Nukem 3D: Megaton Edition I noticed you couldn't run any of it's addons.

The root cause was the run scripts did parse the wrong folder to the engines.
While being already executed from gameroot they added another gameroot to the launch parameters for both engines, raze and eduke32.

This made them look up `gameroot/gameroot/addons/[dc/nw/vacation]` for the game files instead of a single gameroot.

This is fixed now #560 :)

d10sfan reviewed (changes requested):
A few items to change here:

* Normally, it's preferred that only one engine be changed per pull request. That's because it's looking for changes for one engine at a time to do the build process. So if you could break this out, that'd be great.
* The scripts being changed do work for the other version of the  game, so changing these could break others. It would make sense to either detect which game it is, or more simply, copy these scripts and create new ones just for this game. The script that should be run is defined in packagesruntime.json under the app id. You can go into the compatibility tools and edit the config.json file to disable that  file being re-written to. As well, you can edit the file in ~/.cache/luxtorpeda/packages.runtime.json. If this part is too much work, then feel free to create the new file and I can point it to the right location.

Other than that, it looks good. Thanks for the PR!

VortexAcherontic commented:
So I reverted all previous changes and created a new run-raze-megaton.sh script and added it to the packagesruntime.json for DN3D Megaton.

The run-raze.sh and run-eduke32.sh should be reverted to their original state.

The patched for eduke32 I will then do in a 2nd PR.

I hope things now checkout this way :)

d10sfan commented:
@VortexAcherontic Looks good thanks! Feel free to open the other pr once you have it.
